File: Notes/View/MainMenu.py
from View.Commands.GetInfo import GetInfo
from View.Commands.AddItem import AddItem
from View.Commands.EditData import EditData
from View.Commands.RemoveItem import RemoveItem
from View.Commands.SaveData import SaveData
from View.Commands.LoadData import LoadData
from View.Commands.Finish import Finish

from View.Commands.SortByHeader import SortByHeader
from View.Commands.SortByDate import SortByDate
from View.Commands.ClearData import ClearData


class MainMenu:

    def __init__(self, consoleUI):
        self.commandList = []
        self.commandList.append(GetInfo(consoleUI))
        self.commandList.append(AddItem(consoleUI))
        self.commandList.append(EditData(consoleUI))
        self.commandList.append(RemoveItem(consoleUI))
        # self.commandList.append(SortByHeader(consoleUI))
        # self.commandList.append(SortByDate(consoleUI))
        # self.commandList.append(ClearData(consoleUI))
        self.commandList.append(SaveData(consoleUI))
        self.commandList.append(LoadData(consoleUI))

        self.commandList.append(Finish(consoleUI))

    def menu(self):
        sb = ""
        for i in range(len(self.commandList)):
            sb += str(i+1)
            sb += ". "
            sb += self.commandList[i].getDescription()
            sb += "\n"
        return sb

    def execute(self, choice):
        command = self.commandList[choice-1]
        command.execute()

    def getSize(self):
        return len(self.commandList)

File: ToyShop/View/MainMenu.java
package View;

import View.Commands.*;

import java.util.ArrayList;
import java.util.List;

public class MainMenu {
    private List<Command> commandList;

    public MainMenu(ConsoleUI consoleUI) {
        commandList = new ArrayList<>();
        commandList.add(new GetToyShopInfo(consoleUI));
        commandList.add(new AddItem(consoleUI));
        commandList.add(new Raffle(consoleUI));
        commandList.add(new GetPrize(consoleUI));
        // commandList.add(new SortByName(consoleUI));
        // commandList.add(new SortByAge(consoleUI));
        // commandList.add(new FindData(consoleUI));
        // commandList.add(new SaveData(consoleUI));
        // commandList.add(new LoadData(consoleUI));
        commandList.add(new Finish(consoleUI));
    }

    public String menu() {
        StringBuilder stringBuilder = new StringBuilder();
        for (int i = 0; i < commandList.size(); i++) {
            stringBuilder.append(i + 1);
            stringBuilder.append(". ");
            stringBuilder.append(commandList.get(i).getDescription());
            stringBuilder.append("\n");
        }
        return stringBuilder.toString();
    }

    public void execute(int choice) {
        Command command = commandList.get(choice - 1);
        command.execute();
    }

    public int getSize() {
        return commandList.size();
    }
}
